crates/core/c8y_smartrest/src/operations.rs

```diff
@@ -48,6 +48,16 @@ pub struct Operations {
     operations_by_trigger: HashMap<String, usize>,
 }
 
+/// depending on which editor you use, temporary files could be created that contain the name of
+/// the file.
+/// this `operation_name_is_valid` fn will ensure that only files that do not contain
+/// any special characters are allowed.
+pub fn is_valid_operation_name(operation: &str) -> bool {
+    operation
+        .chars()
+        .all(|c| c.is_ascii_alphabetic() | c.is_numeric() | c.eq(&'_'))
```

Comment:
A bit surprised that this function rejects even `-`.

Reply:
yes `_` is not alphabetic() or numeric() and our operations are prefixed with `c8y_`

Reply:
ohh thanks!